src/utils.ts

```diff
@@ -1300,7 +1300,7 @@ export function getMongoDBClientEncryption(): {
  * - `-1 = oid1 is less than oid2`
  * - `+0 = oid1 is equal oid2`
  */
-export function compareObjectId(oid1?: ObjectId, oid2?: ObjectId): 0 | 1 | -1 {
+export function compareObjectId(oid1?: ObjectId | null, oid2?: ObjectId | null): 0 | 1 | -1 {
```

Comment:
```suggestion
export function compareObjectId(oid1: ObjectId | null, oid2: ObjectId | null): 0 | 1 | -1 {
```
will oid1 / oid2 ever be undefined?

Reply:
It won't be in our usage but the function would support those inputs cus of the way we check nullish. I've added tests to keep it this way.